modules/widgets/simple-payments.php

```diff
@@ -0,0 +1,330 @@
+Version: 1.0
+Author: Automattic Inc.
+Author URI: http://automattic.com/
+License: GPLv2 or later
+*/
+
+function jetpack_register_widget_simple_payments() {
+	register_widget( 'Simple_Payments_Widget' );
+}
+add_action( 'widgets_init', 'jetpack_register_widget_simple_payments' );
+
+class Simple_Payments_Widget extends WP_Widget {
+	private static $dir              = null;
+	private static $url              = null;
+	private static $product_defaults = null;
+
+	function __construct() {
+		$widget = array(
+			'classname'   => 'simple-payments',
+			'description' => __( 'Add a simple payment button.', 'jetpack' ),
+		);
+
+		parent::__construct(
+			'Simple_Payments_Widget',
+			apply_filters( 'jetpack_widget_name', __( 'Simple Payments', 'jetpack' ) ),
+			$widget
+		);
+
+		self::$dir = trailingslashit( dirname( __FILE__ ) );
+		self::$url = plugin_dir_url( __FILE__ );
+
+		add_action( 'admin_enqueue_scripts', array( __class__, 'enqueue_admin_styles' ) );
+		add_action( 'wp_enqueue_scripts', array( __class__, 'enqueue_widget_styles' ) );
+	}
+
+	public static function enqueue_admin_styles( $hook_suffix ) {
+		if ( 'widgets.php' == $hook_suffix ) {
+			wp_enqueue_style( 'simple-payments-widget-admin', self::$url . '/simple-payments/style-admin.css', array() );
+			wp_enqueue_media();
+			wp_enqueue_script( 'simple-payments-widget-admin', self::$url . '/simple-payments/admin.js', array( 'jquery' ), false, true );
+		}
+	}
+
+	public static function enqueue_widget_styles() {
+		// TODO: This would be nice, but I don't know how this function works.
+		// if ( is_active_widget( false, false, $this->id_base, true ) ) {
+			wp_enqueue_style( 'simple-payments-widget', self::$url . '/simple-payments/style.css', array() );
+		// }
+	}
+
+	protected static function get_product_defaults() {
+		if ( ! isset( self::$product_defaults ) ) {
+			$products = get_posts( array(
+				'numberposts' => 1,
+				'orderby' => 'date',
+				'post_type' => Jetpack_Simple_Payments::$post_type_product,
+			) );
+			$current_user = wp_get_current_user();
+			$default_email = $current_user->user_email;
+			$default_multiple = '0';
+			$default_currency = 'USD';
+			// Get the default email, currency and "multiple" flag from the last product created, for user's convenience
+			if ( ! empty( $products ) ) {
+				$default_email = get_post_meta( $products[0]->ID, 'spay_email', true );
+				$default_multiple = get_post_meta( $products[0]->ID, 'spay_multiple', true );
+				$default_currency = get_post_meta( $products[0]->ID, 'spay_currency', true );
+			}
+			self::$product_defaults = array( $default_email, $default_multiple, $default_currency );
+		}
+		return self::$product_defaults;
+	}
+
+	protected function get_product_args( $product_id ) {
+		$product = $product_id ? get_post( $product_id ) : null;
+		$product_args = array();
+		if ( $product && ! is_wp_error( $product ) && $product->post_type === Jetpack_Simple_Payments::$post_type_product ) {
```

Comment:
We should probably check and make sure `Jetpack_Simple_Payments` is loaded before to use it, since it can be unloaded with filters right now.